lib/nerves_network/utils.ex

```diff
@@ -17,17 +21,21 @@ defmodule Nerves.Network.Utils do
     end)
   end
 
-  def generate_link_local(mac_address) do
+  def generate_link_local(mac_address, available_subnets_list) do
     <<x, y, _rest :: bytes>> = :crypto.hash(:md5, mac_address)
-    x =
-      case x do
-        255 -> 254
-        0 -> 1
-        v -> v
-      end
+    Enum.at(available_subnets_list, x, Enum.random(available_subnets_list))
     "169.254.#{x}.#{y}"
   end
 
+  def generate_link_local(mac_address, nil) do
```

Comment:
You will probably get a warning about this case never being reached.

Reply:
From dialyzer maybe this is being configured in config.exs though. If not specified there the (mac_address, nil) clause is being onvoked.

Reply:
But because line 22 has no matching on the second argument, I think it will always go to the case at line 22. Not saying to remove it, just move its position above line 22.

Reply:
if not specified, wouldn't you get [] though?

Reply:
![image](https://user-images.githubusercontent.com/15139000/176188476-82a34656-751e-43ce-b19a-eb50fa2f8664.png)

Reply:
resolved...